docking.js

```diff
@@ -1772,14 +1776,9 @@ var DockManager = class DashToDock_DockManager {
         }
 
         this._preferredMonitorIndex = this._settings.get_int('preferred-monitor');
-        // In case of multi-monitor, we consider the dock on the primary monitor to be the preferred (main) one
-        // regardless of the settings
-        // The dock goes on the primary monitor also if the settings are incosistent (e.g. desired monitor not connected).
-        if (this._settings.get_boolean('multi-monitor') ||
-            this._preferredMonitorIndex < 0 || this._preferredMonitorIndex > Main.layoutManager.monitors.length - 1
-            ) {
-            this._preferredMonitorIndex = Main.layoutManager.primaryIndex;
-        } else {
+        if (this._preferredMonitorIndex == -2) {
+            this._preferredMonitorIndex = Meta.MonitorManager.get().get_monitor_for_connector(this._settings.get_string('preferred-monitor-by-connector'))
```

Comment:
```suggestion
            this._preferredMonitorIndex = Meta.MonitorManager.get().get_monitor_for_connector(
                this._settings.get_string('preferred-monitor-by-connector'))
```

Reply:
I think it's fine not to check for error here as in case it will just return `-1`, and we'll proceed as expected.